Lib/test/test_array.py

```diff
@@ -1014,6 +1014,22 @@ def test_pop(self):
             array.array(self.typecode, self.example[3:]+self.example[:-1])
         )
 
+    def test_clear(self):
+        a = array.array(self.typecode, self.example)
+        self.assertRaises(TypeError, a.clear, 42)
+        a.clear()
+        self.assertEqual(a, array.array(self.typecode))
```

Comment:
This is a bit of an indirect test, why not instead test that `len(a) == 0`?

Reply:
All the testing logic from the `test_array.py` file works so that the methods are tested for all possible `typecode`.

To be honest, initially I used array comparison expecting that there would be a `typecode` check, but then I looked at `array_richcompare() `and realized that comparing arrays using the `==` operator is not suitable for checking the same `typecode`.

Therefore, I added two checks - the `len()` you suggested and the `typecode` comparison (just in case of future optimizations).